code/modules/shuttle/custom_shuttle.dm

```diff
@@ -216,10 +184,33 @@
 			to_chat(usr, "<span class='notice'>Unable to comply.</span>")
 	return
 
-/obj/machinery/computer/custom_shuttle/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock, idnum, override=FALSE)
-	if(port && (shuttleId == initial(shuttleId) || override))
+/obj/machinery/computer/system_map/custom_shuttle/proc/can_jump(distance)
+	var/obj/docking_port/mobile/linkedShuttle = SSshuttle.getShuttle(shuttle_id)
+	if(!linkedShuttle)
+		return FALSE
+	if(linkedShuttle.mode != SHUTTLE_IDLE)
+		return FALSE
+	//Calculate our speed
+	if(!calculateStats(FALSE, 0, TRUE))
+		return FALSE
```

Comment:
Can you roll these into one line of checks?

Reply:
Sure, I think it's more elegant spread out